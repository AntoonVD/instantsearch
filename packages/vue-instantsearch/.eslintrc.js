module.exports = {
  extends: 'algolia/vue',
  "rules": {
    "no-warning-comments": "warn", // we have many Todo:, this will remind us to deal with them
  }
};
