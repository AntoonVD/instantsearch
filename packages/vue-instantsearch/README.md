Algolia components for Vue.js
-----------------------------

This project uses [Lerna](https://github.com/lerna/lerna) to orchestrate all the packages.

To get started you should:

```sh
# Install dependencies common to the packages
yarn install

# Download all dependencies and build all packages
yarn boot

# Build the dependencies of all packages
yarn build

# Clean all the dependencies and build again
yarn reboot

# Launch tests with Jest
yarn test
```
