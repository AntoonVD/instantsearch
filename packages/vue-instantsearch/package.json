{
  "name": "vue-instantsearch",
  "author": {
    "name": "Algolia, Inc.",
    "url": "https://www.algolia.com"
  },
  "description": "ðŸ‘€ Lightning-fast Algolia search for Vue apps",
  "keywords": [
    "vue",
    "algolia",
    "search",
    "instantsearch",
    "components",
    "ui",
    "facet",
    "autocomplete"
  ],
  "license": "MIT",
  "version": "0.1.0",
  "files": [
    "dist"
  ],
  "main": "dist/vue-instantsearch.common.js",
  "module": "dist/vue-instantsearch.esm.js",
  "repository": "https://github.com/algolia/vue-instantsearch",
  "scripts": {
    "build": "yarn run build:cjs && yarn run build:es && yarn run build:umd",
    "build:cjs": "rollup -c ./build/rollup.cjs.config.js",
    "build:cjs:watch": "yarn run build:cjs -- -w",
    "build:es": "rollup -c ./build/rollup.es.config.js",
    "build:umd": "rollup -c ./build/rollup.umd.config.js",
    "watch": "concurrently --kill-others \"yarn run test:watch\" \"yarn run build:cjs:watch\"",
    "test": "yarn run lint && jest --verbose --coverage",
    "test:update": "jest --verbose --updateSnapshot",
    "test:watch": "jest --verbose --watchAll --notify",
    "lint": "eslint --ext .js,.vue .",
    "lint:fix": "eslint . --fix"
  },
  "dependencies": {
    "algoliasearch": "^3.18.1",
    "algoliasearch-helper": "^2.14.0",
    "escape-html": "^1.0.3"
  },
  "peerDependencies": {
    "vue": "^2.2.2"
  },
  "devDependencies": {
    "babel-jest": "^19.0.0",
    "babel-preset-es2015": "^6.24.0",
    "concurrently": "^3.4.0",
    "eslint": "^3.19.0",
    "eslint-plugin-html": "^2.0.1",
    "eslint-plugin-prettier": "^2.0.1",
    "jest": "^20.0.3",
    "jest-serializer-html": "^4.0.0",
    "prettier": "^0.22.0",
    "rollup": "^0.41.5",
    "rollup-plugin-buble": "^0.15.0",
    "rollup-plugin-commonjs": "^7.1.0",
    "rollup-plugin-filesize": "^1.0.1",
    "rollup-plugin-json": "^2.1.1",
    "rollup-plugin-node-resolve": "^2.0.0",
    "rollup-plugin-replace": "^1.1.1",
    "rollup-plugin-uglify": "^1.0.1",
    "rollup-plugin-vue": "^2.3.1",
    "rollup-watch": "^3.2.2",
    "vue": "^2.2.2"
  },
  "jest": {
    "collectCoverageFrom": [
      "dist/vue-instantsearch.common.js"
    ],
    "snapshotSerializers": [
      "jest-serializer-html"
    ]
  }
}
