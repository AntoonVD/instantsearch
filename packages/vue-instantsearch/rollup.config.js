import vue from 'rollup-plugin-vue';
import buble from 'rollup-plugin-buble';
import filesize from 'rollup-plugin-filesize';
import resolve from 'rollup-plugin-node-resolve';
import commonjs from 'rollup-plugin-commonjs';
import uglify from 'rollup-plugin-uglify';
import replace from 'rollup-plugin-replace';
import json from 'rollup-plugin-json';

export default [
  {
    input: 'src/instantsearch.js',
    external: [
      'algoliasearch/lite',
      'algoliasearch-helper',
      'escape-html',
      'lodash/range',
      'instantsearch.js/es/',
      'instantsearch.js/es/connectors',
      'instantsearch.js/es/widgets/configure/configure.js',
    ],
    output: [
      {
        file: `dist/vue-instantsearch.common.js`,
        format: 'cjs',
        exports: 'named',
      },
      {
        file: `dist/vue-instantsearch.esm.js`,
        format: 'es',
      },
    ],
    plugins: [
      vue({ compileTemplate: true, css: false }),
      json(),
      buble({
        transforms: {
          dangerousForOf: true,
        },
      }),
      uglify(),
      filesize(),
    ],
  },
  {
    input: 'src/instantsearch.umd.js',
    output: [
      {
        file: `dist/vue-instantsearch.js`,
        format: 'umd',
        name: 'VueInstantSearch',
        exports: 'named',
      },
    ],
    plugins: [
      vue({ compileTemplate: true, css: false }),
      json(),
      resolve({
        browser: true,
        preferBuiltins: false,
      }),
      buble({
        transforms: {
          dangerousForOf: true,
        },
      }),
      commonjs(),
      replace({
        'process.env': JSON.stringify({
          NODE_ENV: 'production',
        }),
      }),
      uglify(),
      filesize(),
    ],
  },
];
