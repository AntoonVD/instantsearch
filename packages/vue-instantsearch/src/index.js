import Index from 'vue-instantsearch-index';
import Highlight from 'vue-instantsearch-highlight';
import Snippet from 'vue-instantsearch-snippet';
import Input from 'vue-instantsearch-input';
import Results from 'vue-instantsearch-results';
import Stats from 'vue-instantsearch-stats';
import Pagination from 'vue-instantsearch-pagination';
import ResultsPerPageSelector
  from 'vue-instantsearch-results-per-page-selector';
import TreeMenu from 'vue-instantsearch-tree-menu';
import SortBySelector from 'vue-instantsearch-sort-by-selector';
import SearchBox from 'vue-instantsearch-search-box';
import Clear from 'vue-instantsearch-clear';
import Rating from 'vue-instantsearch-rating';
import NoResults from 'vue-instantsearch-no-results';
import RefinementList from 'vue-instantsearch-refinement-list';
import PriceRange from 'vue-instantsearch-price-range';
import PoweredBy from 'vue-instantsearch-powered-by';

const InstantSearch = {
  Index,
  Highlight,
  Snippet,
  Input,
  Results,
  Stats,
  Pagination,
  ResultsPerPageSelector,
  TreeMenu,
  SortBySelector,
  SearchBox,
  Clear,
  Rating,
  NoResults,
  RefinementList,
  PriceRange,
  PoweredBy,

  install(Vue) {
    Vue.component('ais-index', Index);
    Vue.component('ais-highlight', Highlight);
    Vue.component('ais-snippet', Snippet);
    Vue.component('ais-input', Input);
    Vue.component('ais-results', Results);
    Vue.component('ais-stats', Stats);
    Vue.component('ais-pagination', Pagination);
    Vue.component('ais-results-per-page-selector', ResultsPerPageSelector);
    Vue.component('ais-tree-menu', TreeMenu);
    Vue.component('ais-sort-by-selector', SortBySelector);
    Vue.component('ais-search-box', SearchBox);
    Vue.component('ais-clear', Clear);
    Vue.component('ais-rating', Rating);
    Vue.component('ais-no-results', NoResults);
    Vue.component('ais-refinement-list', RefinementList);
    Vue.component('ais-price-range', PriceRange);
    Vue.component('ais-powered-by', PoweredBy);
  },
};

// Automatically register Algolia Search components if Vue is available globally
if (typeof window !== 'undefined' && window.Vue) {
  window.Vue.use(InstantSearch);
}

export default InstantSearch;

export {
  Index,
  Highlight,
  Snippet,
  Input,
  Results,
  Stats,
  Pagination,
  ResultsPerPageSelector,
  TreeMenu,
  SortBySelector,
  SearchBox,
  Clear,
  Rating,
  NoResults,
  RefinementList,
  PriceRange,
  PoweredBy,
};
