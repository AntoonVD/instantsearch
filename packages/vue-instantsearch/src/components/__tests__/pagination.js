import Vue from 'vue';
import { Pagination } from 'vue-instantsearch';

test('renders proper HTML', () => {
  const searchStore = {
    page: 2,
    totalPages: 10,
  };
  const Component = Vue.extend(Pagination);
  const vm = new Component({
    propsData: {
      searchStore,
    },
  });
  vm.$mount();

  expect(vm.$el.outerHTML).toMatchSnapshot();
});

test('accepts custom padding', () => {
  const searchStore = {
    page: 5,
    totalPages: 10,
  };
  const Component = Vue.extend(Pagination);
  const vm = new Component({
    propsData: {
      padding: 2,
      searchStore,
    },
  });
  vm.$mount();

  expect(vm.$el.outerHTML).toMatchSnapshot();
});

test('it should not try to go to a previous page that would be inferior to 1', () => {
  const searchStore = {
    page: 1,
    totalPages: 20,
  };
  const Component = Vue.extend(Pagination);
  const vm = new Component({
    propsData: {
      searchStore,
    },
  });

  vm.goToPreviousPage();

  expect(searchStore.page).toEqual(1);
});

test('it should not try to go to a next page that would be superior to total existing pages', () => {
  const searchStore = {
    page: 20,
    totalPages: 20,
  };
  const Component = Vue.extend(Pagination);
  const vm = new Component({
    propsData: {
      searchStore,
    },
  });

  vm.goToNextPage();

  expect(searchStore.page).toEqual(20);
});
