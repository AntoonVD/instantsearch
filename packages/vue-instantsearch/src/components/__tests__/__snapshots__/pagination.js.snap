// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts custom padding 1`] = `

<ul class="ais-pagination">
  <li class="ais-pagination__item ais-pagination__item--first">
    <a href="#"
       class="ais-pagination__link"
    >
      &lt;&lt;
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item--previous">
    <a href="#"
       class="ais-pagination__link"
    >
      &lt;
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      3
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      4
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item ais-pagination__item--active">
    <a href="#"
       class="ais-pagination__link"
    >
      5
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      6
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      7
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item--next">
    <a href="#"
       class="ais-pagination__link"
    >
      &gt;
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item--last">
    <a href="#"
       class="ais-pagination__link"
    >
      &gt;&gt;
    </a>
  </li>
</ul>

`;

exports[`it should be hidden if there are no results in the current context 1`] = `

<ul class="ais-pagination"
    style="display: none;"
>
  <li class="ais-pagination__item ais-pagination__item--first ais-pagination__item ais-pagination__item--disabled">
    <a href="#"
       class="ais-pagination__link"
    >
      &lt;&lt;
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item--previous ais-pagination__item ais-pagination__item--disabled">
    <a href="#"
       class="ais-pagination__link"
    >
      &lt;
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item ais-pagination__item--active">
    <a href="#"
       class="ais-pagination__link"
    >
      1
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      2
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      3
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      4
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      5
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      6
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      7
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item--next">
    <a href="#"
       class="ais-pagination__link"
    >
      &gt;
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item--last">
    <a href="#"
       class="ais-pagination__link"
    >
      &gt;&gt;
    </a>
  </li>
</ul>

`;

exports[`renders proper HTML 1`] = `

<ul class="ais-pagination">
  <li class="ais-pagination__item ais-pagination__item--first">
    <a href="#"
       class="ais-pagination__link"
    >
      &lt;&lt;
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item--previous">
    <a href="#"
       class="ais-pagination__link"
    >
      &lt;
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      1
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item ais-pagination__item--active">
    <a href="#"
       class="ais-pagination__link"
    >
      2
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      3
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      4
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      5
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      6
    </a>
  </li>
  <li class="ais-pagination__item">
    <a href="#"
       class="ais-pagination__link"
    >
      7
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item--next">
    <a href="#"
       class="ais-pagination__link"
    >
      &gt;
    </a>
  </li>
  <li class="ais-pagination__item ais-pagination__item--last">
    <a href="#"
       class="ais-pagination__link"
    >
      &gt;&gt;
    </a>
  </li>
</ul>

`;
